### PR TITLE
add webmanifest's shortcuts support

### PR DESCRIPTION
**Pull request by chrmoritz**
This PR adds support for webmanifest's shortcuts. For more information see:

https://developer.mozilla.org/en-US/docs/Web/Manifest/shortcuts
https://web.dev/app-shortcuts/
https://w3c.github.io/manifest/#shortcuts-member

Example usage:
```js
const result = await favicons(logo_png, {
  shortcuts: [
    {
      name: "View your Inbox",
      short_name: "inbox",
      description: "View your inbox messages",
      url: "/inbox",
      icon: ["test/shortcut1.png", "test/shortcut1-small.png"],
    },
    {
      name: "Picture Gallery",
      short_name: "pictures",
      description: "View your own Picture Gallery",
      url: "/pictures",
      icon: "test/shortcut2.svg"
    },
  ],
});
```

See also the added test and readme changes for usage of this new feature.

### CODE REVIEW

File: src/config/defaults.ts [hunk 1]
```diff
@@ -19,6 +19,14 @@ export interface FileOptions {
   readonly manifestFileName?: string;
 }
 
+export interface ShortcutOptions {
+  readonly name: string;
+  readonly short_name?: string;
+  readonly description?: string;
+  readonly url: string;
+  readonly icon?: string | Array<string>;
```
Comment (andy128k) [inline src/config/defaults.ts:27]: There is no style guide in this package. But all arrays are written as `type[]`.
It there a good reason to prefer one syntax over another?

File: src/config/defaults.ts [hunk 2]
```diff
@@ -57,6 +65,7 @@ export interface FaviconOptions {
   readonly relatedApplications?: Application[];
   readonly icons?: Dictionary<IconOptions | boolean | string[]>;
   readonly files?: Dictionary<FileOptions>;
+  readonly shortcuts?: Dictionary<ShortcutOptions>;
```
Comment (andy128k) [inline src/config/defaults.ts:68]: Type is `Dictionary<>` but all examples use arrays. It seems like correct type should be `ShortcutOptions[]`.

File: src/platforms/android.ts [hunk 3]
```diff
@@ -75,6 +84,28 @@ export class AndroidPlatform extends Platform {
...
+          const shortcutIcons = await Promise.all(
+            Object.entries(SHORTCUT_ICONS_OPTIONS).map(
+              ([shortcutName, option]) =>
+                images.createFavicon(
+                  shortcutSourceset,
+                  `shortcut${index + 1}-${shortcutName}`,
+                  option
+                )
+            )
+          );
```
Comment (andy128k) [inline src/platforms/android.ts:104]: Code is pretty complex, so, I think it deserves to have it's own method.

File: src/platforms/android.ts [hunk 4]
```diff
@@ -163,6 +194,36 @@ export class AndroidPlatform extends Platform {
...
+                    basePath,
+                    `shortcut${index + 1}-${shortcutName}`
+                  ),
+                  sizes: `${width}x${height}`,
+                  type: "image/png",
+                };
+              }
+            );
+          }
+          return result;
```
Comment (andy128k) [inline src/platforms/android.ts:222]: Same here. I seems natural to move this to its own method.

**Review by andy128k — COMMENTED** [review]
I think this feature should respect option `output: { images: false }`.

File: src/config/defaults.ts (hunk 2 above)
Comment (chrmoritz) [reply src/config/defaults.ts]: Oops forget to change that.

**Comment by chrmoritz** [comment]
Thank you for the fast review, I've addressed all your comments and fixed another typo from me, which broke `short_name`.

File: src/platforms/android.ts [hunk 5]
```diff
@@ -91,6 +120,26 @@ export class AndroidPlatform extends Platform {
...
+            images.createFavicon(
+              shortcutSourceset,
+              `shortcut${index + 1}-${shortcutName}`,
+              option
+            )
+          )
+        );
+      })
+    );
+    return icons.flat();
```
Comment (chrmoritz) [inline src/platforms/android.ts:140]: Just realized that this isn't supported in Node.js 10. Are you happy to require Node.js 12+ in v7 and dropping support for this EOL Node.js version or do I need to rewrite this?

File: readme.md [hunk 6]
```diff
@@ -64,6 +64,18 @@ var favicons = require("favicons"),
...
     },
+    shortcuts: [
+      // Your applications's Shortcuts (see: https://developer.mozilla.org/docs/Web/Manifest/shortcuts)
+      // Array of shortcut objects:
+      {
+        name: "View your Inbox", // The name of the shortcut. `string`
+        short_name: "inbox", // optionally, falls back to name. `string`
+        description: "View your inbox messages", // optionally, not used in any implemention yet. `string`
+        url: "/inbox", // The URL this shortcut should lead to. `string`
+        icon: "test/inbox_shortcut.png", // source image(s) for that shortcut. `string`, `buffer` or array of `string`
```
Comment (andy128k) [inline readme.md:75]: This comment contradicts to a type.

File: src/config/defaults.ts [hunk 7]
```diff
@@ -19,6 +19,14 @@ export interface FileOptions {
   readonly manifestFileName?: string;
 }
 
+export interface ShortcutOptions {
+  readonly name: string;
+  readonly short_name?: string;
+  readonly description?: string;
+  readonly url: string;
+  readonly icon?: string | string[];
```
Comment (andy128k) [inline src/config/defaults.ts:27]: `Buffer` was mentioned in a comment. I suppose we may allow it too. 

`string | string[] | Buffer | Buffer[]` should be ok.

File: src/platforms/android.ts (hunk 5 above)
Comment (andy128k) [reply src/platforms/android.ts:140]: That's totally OK.

**Review by andy128k — CHANGES_REQUESTED** [review]

**Comment by andy128k** [comment]
It looks like both maskable and shortcuts respect `output: { images: false }`.

File: src/config/defaults.ts (hunk 7 above)
Comment (chrmoritz) [reply src/config/defaults.ts]: Right, but the type of `manifestMaskable`contradicts the readme in that point too.
I've also fixed that.

File: test/manifestShortcuts.test.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
+import favicons from "..";
+import * as fs from "fs";
+import { logo_png, logo_svg } from "./util";
+
+test("manifest should support shortcuts", async () => {
+  expect.assertions(1);
+
+  const result = await favicons(logo_png, {
```
Comment (andy128k) [inline test/manifestShortcuts.test.js:8]: May you also add 

```
    icons: {
      android: true,
      appleIcon: false,
      appleStartup: false,
      favicons: true,
      windows: false,
      yandex: false,
    },
```
to test, so we do now blow snapshots.

Comment (andy128k) [inline test/manifestShortcuts.test.js:8]: May you also add 

```
    icons: {
      android: true,
      appleIcon: false,
      appleStartup: false,
      favicons: false,
      windows: false,
      yandex: false,
    },
```
to test, so we do now blow snapshots.

Comment (chrmoritz) [reply test/manifestShortcuts.test.js:8]: make sense

File: src/platforms/android.ts [hunk 9]
```diff
@@ -11,6 +11,20 @@ import {
...
+interface Icon {
+  readonly src: string;
+  readonly sizes: string;
+  readonly type: string;
+}
+
+interface Shortcut {
+  readonly name: string;
+  readonly short_name: string;
+  description?: string;
```
Comment (andy128k) [inline src/platforms/android.ts:23]:
```suggestion
  readonly description?: string;
```

File: src/platforms/android.ts [hunk 10]
```diff
@@ -11,6 +11,20 @@ import {
...
+  readonly url: string;
+  icons?: Icon[];
```
Comment (andy128k) [inline src/platforms/android.ts:25]:
```suggestion
  readonly icons?: Icon[];
```

File: src/platforms/android.ts (hunk 9 above)
Comment (chrmoritz) [reply src/platforms/android.ts]: done
I needed to do a little rewrite for that though.

**Review by andy128k — APPROVED** [review]

**Comment by andy128k** [comment]
@chrmoritz Thanks!

**Pull request merged**